src/kv/bulk.rs

```diff
@@ -33,36 +39,106 @@ pub fn put(
     target: &Target,
     user: &GlobalUser,
     namespace_id: &str,
-    pairs: &[KeyValuePair],
+    pairs: Vec<KeyValuePair>,
+    progress_bar: &Option<ProgressBar>,
 ) -> Result<(), failure::Error> {
     let client = bulk_api_client(user)?;
 
-    match client.request(&WriteBulk {
-        account_identifier: &target.account_id,
-        namespace_identifier: namespace_id,
-        bulk_key_value_pairs: pairs.to_owned(),
-    }) {
-        Ok(_) => Ok(()),
-        Err(e) => failure::bail!("{}", format_error(e)),
+    for b in batch_keys_values(pairs) {
+        match client.request(&WriteBulk {
+            account_identifier: &target.account_id,
+            namespace_identifier: namespace_id,
+            bulk_key_value_pairs: b.to_owned(),
+        }) {
+            Ok(_) => {}
+            Err(e) => failure::bail!("{}", format_error(e)),
+        }
+
+        if let Some(pb) = &progress_bar {
+            pb.inc(b.len() as u64);
+        }
     }
+
+    Ok(())
 }
 
 pub fn delete(
     target: &Target,
     user: &GlobalUser,
     namespace_id: &str,
     keys: Vec<String>,
+    progress_bar: &Option<ProgressBar>,
 ) -> Result<(), failure::Error> {
     let client = bulk_api_client(user)?;
 
-    let response = client.request(&DeleteBulk {
-        account_identifier: &target.account_id,
-        namespace_identifier: namespace_id,
-        bulk_keys: keys,
-    });
+    for b in batch_keys(keys) {
+        match client.request(&DeleteBulk {
+            account_identifier: &target.account_id,
+            namespace_identifier: namespace_id,
+            bulk_keys: b.to_owned(),
+        }) {
+            Ok(_) => {}
+            Err(e) => failure::bail!("{}", format_error(e)),
+        }
+
+        if let Some(pb) = &progress_bar {
+            pb.inc(b.len() as u64);
+        }
+    }
+
+    Ok(())
+}
+
+fn batch_keys_values(mut pairs: Vec<KeyValuePair>) -> Vec<Vec<KeyValuePair>> {
+    let mut batches: Vec<Vec<KeyValuePair>> = Vec::new();
 
-    match response {
-        Ok(_) => Ok(()),
-        Err(e) => failure::bail!("{}", format_error(e)),
+    if !pairs.is_empty() {
```

Comment:
this if is probably unnecessary; it just saves initializing a few local variables. felt cute might delete later.